harness/executor/harness.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"sync"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/lyft/protoc-gen-validate/harness"
+	"golang.org/x/net/context"
+)
+
+var Harnesses = []Harness{
+	InitHarness("harness/go/go-harness"),
+}
+
+type Harness func(context.Context, io.Reader) (*harness.TestResult, error)
+
+func InitHarness(cmd string, args ...string) Harness {
+	return func(ctx context.Context, r io.Reader) (*harness.TestResult, error) {
+		out, errs := getBuf(), getBuf()
+		defer relBuf(out)
```

Comment:
There's no way in Go to both assign and defer at the same time? I'm thinking of something like RAII or Python contexts.

Reply:
Unfortunately no :/